### DISCUSSION TITLE
TP3 - Duda ejercicio 1.3

### DISCUSSION THREAD

**Discussion by luciano-repetti** [INGCOM-UNRN-P1/cursada-2023#102] [Preguntas y respuestas] [answered]
Buenas! En ejercicio 1.1 dice que desarrolle una función que por argumento le pasemos las capacidad que tendrá y el arreglo a rellenar. El problema con el que me encuentro es que hago un printf() y scanf() en la funcion para cargar los valores que se ponen por teclado en cada posición:

```C
int cargar_arreglo(int capacidad, int arreglo[])
{
    for (int i = 0; i < capacidad; i++)
    {
        int valor_en_posicion;
        printf("Ingrese un numero que ira en la posicion %d del array: ", i);
        scanf("%d", &valor_en_posicion);
        arreglo[i] = valor_en_posicion;
    }
}
``` 

La pregunta es si esta bien hacer el printf() en la función.
Además que en el "ejercicio 1.2" hago un printf() para mostrar el contenido.
Y vi que el profe Martin le respondió a un compañero que en el ejercicio 1.3 se tenia que reutilizar la función cargador de arreglos para fusionar dos arrays.

**Comment by martinvilu** [comment]
usá un `fflush(stdout);` entre el `printf` y el `scanf`; o agrega un `\n`

**Comment by martinvilu** [comment, marked as answer]
Si el proposito de la función incluye mostrar, entonces sí.

En cualquier otro caso no, porque cambia el objetivo de la función.
